graphene_mongo/utils.py

```diff
@@ -101,3 +102,61 @@ def get_node_from_global_id(node, info, global_id):
                 return interface.get_node_from_global_id(info, global_id)
     except AttributeError:
         return Node.get_node_from_global_id(info, global_id)
+
+
+def collect_query_fields(node, fragments):
+    """Recursively collects fields from the AST
+
+    Args:
+        node (dict): A node in the AST
+        fragments (dict): Fragment definitions
+
+    Returns:
+        A dict mapping each field found, along with their sub fields.
+
+        {'name': {},
+         'image': {'id': {},
+                                   'name': {},
+                                   'description': {}},
+         'slug': {}}
+    """
+
+    field = {}
+
+    if node.get('selection_set'):
+        for leaf in node['selection_set']['selections']:
+            if leaf['kind'] == 'Field':
+                field.update({
+                    leaf['name']['value']: collect_query_fields(leaf, fragments)
+                })
+            elif leaf['kind'] == 'FragmentSpread':
+                field.update(collect_query_fields(fragments[leaf['name']['value']],
+                                                  fragments))
+
+    return field
+
+
+def get_query_fields(info):
+    """A convenience function to call collect_query_fields with info
+
+    Args:
+        info (ResolveInfo)
+
+    Returns:
+        dict: Returned from collect_query_fields
+    """
+
+    fragments = {}
+    node = ast_to_dict(info.field_asts[0])
+
+    for name, value in info.fragments.items():
+        fragments[name] = ast_to_dict(value)
+
+    query = collect_query_fields(node, fragments)
+    if "edges" in query:
+        return query["edges"]["node"].keys()
+    return query
+
+
+def camel_to_snake(field):
+    return ''.join(['_' + c.lower() if c.isupper() else c for c in field]).lstrip('_')
```

Comment:
@arunsureshkumar : 
Thanks for this pr, and I think you can use `to_snake` from graphene instead of a new one.
https://github.com/graphql-python/graphene/blob/a9625dac0ed07d5fb35adc83cbc41387755f080f/graphene/utils/str_converters.py